### PR TITLE
Documentation Pull Request

### PR DESCRIPTION
**Pull request by hemilpanchiwala**
This PR
- updates the complete documentation of SBSCL
- updates the site files of SBSCL
- adds UserGuidelines to use the features of SBSCL
- adds sequence diagrams for the developers to understand the workflow of SBSCL
- updates README.md

Many small updates have been made with the above-mentioned points in the PR.

### CODE REVIEW

File: src/main/java/org/simulator/examples/CompExample.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public static void main(String[] args) throws XMLStreamException,
     File file = new File(args[0]);
 
     CompSimulator compSimulator = new CompSimulator(file);
-    MultiTable solution = compSimulator.solve(stepSize = 0.1, timeEnd = 100.0);
+    MultiTable solution = compSimulator.solve(timeEnd = 100.0, stepSize = 0.1);
```
Comment (draeger) [inline src/main/java/org/simulator/examples/CompExample.java:77]: As always, could we please use `100d` and `0.1d`?

**Review by draeger — CHANGES_REQUESTED** [review]

File: UserGuidelines.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+```java
+// An example of solving with Rosenbrock solver
+DESSolver solver = new RosenbrockSolver();
+solver.setStepSize(stepSize); // Setting the step size for the model
+MultiTable solution = solver.solve(interpreter, 
+              interpreter.getInitialValues(), 0d, timeEnd, simulatorExample);
+```
+
+- You can now print the [MultiTable](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/math/odes/MultiTable.java) or can plot it using the [PlotMultiTable](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/plot/PlotMultiTable.java) where you can see the changing values in a graphical form. Also, you can view it in a tabular form using the JTable class.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:28]:
```suggestion
- One can now print the [MultiTable](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/math/odes/MultiTable.java) or can plot it using the [PlotMultiTable](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/plot/PlotMultiTable.java) where you can see the changing values in a graphical form. Also, you can view it in a tabular form using the JTable class.
```

File: UserGuidelines.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+**Note:** To set the absolute and relative tolerances for the specific simulation, you can use the method provided by the [DESolver](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/math/odes/DESSolver.java) using the below code snippets:
+```java
+((AdaptiveStepsizeIntegrator) solver).setAbsTol(absTol);
+((AdaptiveStepsizeIntegrator) solver).setRelTol(relTol);
+```
+
+For complete code on how to simulate the SBML model, please refer to the [SimulatorExample](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/examples/SimulatorExample.java) in the repository.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:36]:
```suggestion
For the complete code on how to simulate an SBML model, please refer to the [SimulatorExample](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/examples/SimulatorExample.java) in the repository.
```

File: UserGuidelines.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+## Simulating the SBML models with comp extension
+- Simulating the comp models is quite easy as you just need to provide a file to the [CompSimulator](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/comp/CompSimulator.java), and it performs all the tasks including the initializations and processings.
+
+```java
+CompSimulator compSimulator = new CompSimulator(sbmlfile);
+```
+
+- After creating the instance, you have to call `solve()` method of the CompSimulator class with duration and step size to get the results in the form of MultiTable.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:45]:
```suggestion
- After creating the instance of the simulator, you have to call the`solve()` method of the CompSimulator class with duration and step size to get the results in the form of MultiTable.
```

**Comment by shalinshah1993** [comment]
- Some links are missing in dependencies.html page. For example, `libkisao`, `GLPKSolverPack`, etc.
- In summary.html, there is version 1.5. After the changes should we update at least the minor version? 
- In the current management.html page, the link points to shalinshah1993's SBSCL version. Great to see you fixed that! 

- BTW this is also a good chance to update pictures in team page, if anyone wants to add? For example, we can add @zakharc there

Otherwise ... 
image

File: UserGuidelines.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+```java
+// Here, 10.0 refers to the total duration
+//       0.1 refers to the step size 
+MultiTable solution = compSimulator.solve(10.0, 0.1);
+```
+
+- After this, you can view the results either by printing or by [PlotMultiTable](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/plot/PlotMultiTable.java) (in graphical form) or by JTable (in tabular form).
+
+For complete code on how to simulate the comp model, please refer to the [CompExample](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/examples/CompExample.java) in the repository.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:55]:
```suggestion
For the complete code on how to simulate the comp model, please refer to the [CompExample](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/examples/CompExample.java) in the repository.
```

File: UserGuidelines.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+## Simulating the SBML models with fbc extension
+
+- Similar to the CompSimulator, here we have to provide the [SBMLDocument](https://github.com/sbmlteam/jsbml/blob/master/core/src/org/sbml/jsbml/SBMLDocument.java) by reading from the file to the [FluxBalanceAnalysis](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/fba/FluxBalanceAnalysis.java) class which then performs all the functionalities.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:59]:
```suggestion
- Similar to the CompSimulator, here we have to provide the [SBMLDocument](https://github.com/sbmlteam/jsbml/blob/master/core/src/org/sbml/jsbml/SBMLDocument.java) by reading from the file to the [FluxBalanceAnalysis](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/org/simulator/fba/FluxBalanceAnalysis.java) class which implements all the flux balance analysis functionality.
```

File: UserGuidelines.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+```java
+SBMLDocument document = new SBMLReader().read(sbmlfile);
+FluxBalanceAnalysis solver = new FluxBalanceAnalysis(document);
+```
+
+- After this, you just need to call `solve()` method of FluxBalanceAnalysis that returns a boolean whether the results are processed successfully or not.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:66]:
```suggestion
- After this, you just need to call `solve()` method of FluxBalanceAnalysis that returns a boolean indicating of the simulation was successful.
```

File: UserGuidelines.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+```java
+boolean solvedStatus = solver.solve();
+```
+
+- Once `solve()` method returns true, you can access different results as described below in the snippet.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:72]:
```suggestion
- After solving the FBA problem the simulation results can be accessed via.  
```

File: UserGuidelines.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+``` 
+
+- After creating the network, you need to initialize the [Simulator](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/fern/simulation/Simulator.java) with the algorithm you wish to simulate by passing the SBMLNetwork instance.
+
+```java
+// Initializes simulator with the GillespieEnhanced Algorithm
+Simulator sim = new GillespieEnhanced(net);
+```
+
+All the supported algorithms for stochastic simulation are present under the [/java/fern/simulation/algorithm](https://github.com/draeger-lab/SBSCL/tree/master/src/main/java/fern/simulation/algorithm) directory.
```
Comment (matthiaskoenig) [inline UserGuidelines.md:106]:
```suggestion
All supported algorithms for stochastic simulation are available in the [/java/fern/simulation/algorithm](https://github.com/draeger-lab/SBSCL/tree/master/src/main/java/fern/simulation/algorithm) directory.
```

File: UserGuidelines.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+**Note:** If your SBML models contain any events, then the network has to call `registerEvents()` method passing the simulator as to keep track of event properties like trigger, delays, and others by the [SBMLEventHandlerObserver](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/fern/network/sbml/SBMLEventHandlerObserver.java).
```
Comment (matthiaskoenig) [inline UserGuidelines.md:108]:
```suggestion
**Note:** If your SBML model contains any events, then the network has to call `registerEvents()` passing the simulator as to keep track of event properties like trigger, delays, and others by the [SBMLEventHandlerObserver](https://github.com/draeger-lab/SBSCL/blob/master/src/main/java/fern/network/sbml/SBMLEventHandlerObserver.java).
```

**Review by matthiaskoenig — APPROVED** [review]
Some smaller wording comments. Otherwise this looks very good.

**Comment by hemilpanchiwala** [comment]
> Some links are missing in dependencies.html page. For example, libkisao, GLPKSolverPack, etc.

@shalinshah1993, the links are generated automatically by the command `mvn site`. But the links are not generated for the 4 libraries as they are used in maven from the `lib` folder only (they are present in repository only and others are from maven).

**Comment by matthiaskoenig** [comment]
Please add the diagrams to the User Guidelines. I don't see the diagrams anywhere displayed. These should be part of the documentation and not only exist in a folder.

**Pull request merged**